# Comparing `tmp/cnmfsns-1.4.1.tar.gz` & `tmp/cnmfsns-1.5.0.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "cnmfsns-1.4.1.tar", last modified: Thu Jun 15 16:16:10 2023, max compression
+gzip compressed data, was "cnmfsns-1.5.0.tar", last modified: Wed Jun 21 20:06:38 2023, max compression
```

## Comparing `cnmfsns-1.4.1.tar` & `cnmfsns-1.5.0.tar`

### file list

```diff
@@ -1,25 +1,25 @@
-drwxrwxrwx   0        0        0        0 2023-06-15 16:16:10.600101 cnmfsns-1.4.1/
--rw-rw-rw-   0        0        0     1090 2023-04-24 20:31:27.000000 cnmfsns-1.4.1/LICENSE
--rw-rw-rw-   0        0        0     3982 2023-06-15 16:16:10.600101 cnmfsns-1.4.1/PKG-INFO
--rw-rw-rw-   0        0        0     3467 2023-06-15 16:15:24.000000 cnmfsns-1.4.1/README.md
--rw-rw-rw-   0        0        0       86 2023-04-24 20:31:27.000000 cnmfsns-1.4.1/pyproject.toml
--rw-rw-rw-   0        0        0      964 2023-06-15 16:16:10.600101 cnmfsns-1.4.1/setup.cfg
-drwxrwxrwx   0        0        0        0 2023-06-15 16:16:10.550087 cnmfsns-1.4.1/src/
-drwxrwxrwx   0        0        0        0 2023-06-15 16:16:10.570242 cnmfsns-1.4.1/src/cnmfsns/
--rw-rw-rw-   0        0        0      513 2023-05-12 18:06:16.000000 cnmfsns-1.4.1/src/cnmfsns/__init__.py
--rw-rw-rw-   0        0        0    51776 2023-06-15 16:15:26.000000 cnmfsns-1.4.1/src/cnmfsns/cli.py
--rw-rw-rw-   0        0        0    39682 2023-05-12 20:51:27.000000 cnmfsns-1.4.1/src/cnmfsns/cnmf.py
--rw-rw-rw-   0        0        0    16061 2023-04-24 20:31:27.000000 cnmfsns-1.4.1/src/cnmfsns/colors.py
--rw-rw-rw-   0        0        0     3257 2023-04-24 20:31:27.000000 cnmfsns-1.4.1/src/cnmfsns/config.py
--rw-rw-rw-   0        0        0    35838 2023-06-14 22:17:36.000000 cnmfsns-1.4.1/src/cnmfsns/dataset.py
--rw-rw-rw-   0        0        0    21163 2023-06-15 03:50:05.000000 cnmfsns-1.4.1/src/cnmfsns/integration.py
--rw-rw-rw-   0        0        0    59336 2023-06-15 03:48:57.000000 cnmfsns-1.4.1/src/cnmfsns/plots.py
--rw-rw-rw-   0        0        0    35301 2023-05-16 19:23:15.000000 cnmfsns-1.4.1/src/cnmfsns/sns.py
--rw-rw-rw-   0        0        0     2767 2023-04-24 20:31:27.000000 cnmfsns-1.4.1/src/cnmfsns/utils.py
-drwxrwxrwx   0        0        0        0 2023-06-15 16:16:10.600101 cnmfsns-1.4.1/src/cnmfsns.egg-info/
--rw-rw-rw-   0        0        0     3982 2023-06-15 16:16:10.000000 cnmfsns-1.4.1/src/cnmfsns.egg-info/PKG-INFO
--rw-rw-rw-   0        0        0      472 2023-06-15 16:16:10.000000 cnmfsns-1.4.1/src/cnmfsns.egg-info/SOURCES.txt
--rw-rw-rw-   0        0        0        1 2023-06-15 16:16:10.000000 cnmfsns-1.4.1/src/cnmfsns.egg-info/dependency_links.txt
--rw-rw-rw-   0        0        0       44 2023-06-15 16:16:10.000000 cnmfsns-1.4.1/src/cnmfsns.egg-info/entry_points.txt
--rw-rw-rw-   0        0        0      177 2023-06-15 16:16:10.000000 cnmfsns-1.4.1/src/cnmfsns.egg-info/requires.txt
--rw-rw-rw-   0        0        0        8 2023-06-15 16:16:10.000000 cnmfsns-1.4.1/src/cnmfsns.egg-info/top_level.txt
+drwxrwxrwx   0        0        0        0 2023-06-21 20:06:38.944065 cnmfsns-1.5.0/
+-rw-rw-rw-   0        0        0     1090 2023-04-24 20:31:27.000000 cnmfsns-1.5.0/LICENSE
+-rw-rw-rw-   0        0        0     4246 2023-06-21 20:06:38.944065 cnmfsns-1.5.0/PKG-INFO
+-rw-rw-rw-   0        0        0     3731 2023-06-21 20:02:01.000000 cnmfsns-1.5.0/README.md
+-rw-rw-rw-   0        0        0       86 2023-04-24 20:31:27.000000 cnmfsns-1.5.0/pyproject.toml
+-rw-rw-rw-   0        0        0      980 2023-06-21 20:06:38.944065 cnmfsns-1.5.0/setup.cfg
+drwxrwxrwx   0        0        0        0 2023-06-21 20:06:38.888950 cnmfsns-1.5.0/src/
+drwxrwxrwx   0        0        0        0 2023-06-21 20:06:38.908788 cnmfsns-1.5.0/src/cnmfsns/
+-rw-rw-rw-   0        0        0      513 2023-05-12 18:06:16.000000 cnmfsns-1.5.0/src/cnmfsns/__init__.py
+-rw-rw-rw-   0        0        0    51994 2023-06-21 19:58:59.000000 cnmfsns-1.5.0/src/cnmfsns/cli.py
+-rw-rw-rw-   0        0        0    39682 2023-05-12 20:51:27.000000 cnmfsns-1.5.0/src/cnmfsns/cnmf.py
+-rw-rw-rw-   0        0        0    16061 2023-04-24 20:31:27.000000 cnmfsns-1.5.0/src/cnmfsns/colors.py
+-rw-rw-rw-   0        0        0     3257 2023-04-24 20:31:27.000000 cnmfsns-1.5.0/src/cnmfsns/config.py
+-rw-rw-rw-   0        0        0    35838 2023-06-14 22:17:36.000000 cnmfsns-1.5.0/src/cnmfsns/dataset.py
+-rw-rw-rw-   0        0        0    21163 2023-06-19 20:28:23.000000 cnmfsns-1.5.0/src/cnmfsns/integration.py
+-rw-rw-rw-   0        0        0    59336 2023-06-15 03:48:57.000000 cnmfsns-1.5.0/src/cnmfsns/plots.py
+-rw-rw-rw-   0        0        0    41117 2023-06-21 18:18:46.000000 cnmfsns-1.5.0/src/cnmfsns/sns.py
+-rw-rw-rw-   0        0        0     2767 2023-04-24 20:31:27.000000 cnmfsns-1.5.0/src/cnmfsns/utils.py
+drwxrwxrwx   0        0        0        0 2023-06-21 20:06:38.944065 cnmfsns-1.5.0/src/cnmfsns.egg-info/
+-rw-rw-rw-   0        0        0     4246 2023-06-21 20:06:38.000000 cnmfsns-1.5.0/src/cnmfsns.egg-info/PKG-INFO
+-rw-rw-rw-   0        0        0      472 2023-06-21 20:06:38.000000 cnmfsns-1.5.0/src/cnmfsns.egg-info/SOURCES.txt
+-rw-rw-rw-   0        0        0        1 2023-06-21 20:06:38.000000 cnmfsns-1.5.0/src/cnmfsns.egg-info/dependency_links.txt
+-rw-rw-rw-   0        0        0       44 2023-06-21 20:06:38.000000 cnmfsns-1.5.0/src/cnmfsns.egg-info/entry_points.txt
+-rw-rw-rw-   0        0        0      189 2023-06-21 20:06:38.000000 cnmfsns-1.5.0/src/cnmfsns.egg-info/requires.txt
+-rw-rw-rw-   0        0        0        8 2023-06-21 20:06:38.000000 cnmfsns-1.5.0/src/cnmfsns.egg-info/top_level.txt
```

### Comparing `cnmfsns-1.4.1/LICENSE` & `cnmfsns-1.5.0/LICENSE`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/PKG-INFO` & `cnmfsns-1.5.0/README.md`

 * *Files 6% similar despite different names*

```diff
@@ -1,249 +1,234 @@
-00000000: 4d65 7461 6461 7461 2d56 6572 7369 6f6e  Metadata-Version
-00000010: 3a20 322e 310d 0a4e 616d 653a 2063 6e6d  : 2.1..Name: cnm
-00000020: 6673 6e73 0d0a 5665 7273 696f 6e3a 2031  fsns..Version: 1
-00000030: 2e34 2e31 0d0a 5375 6d6d 6172 793a 2063  .4.1..Summary: c
-00000040: 4e4d 4620 536f 6c75 7469 6f6e 204e 6574  NMF Solution Net
-00000050: 776f 726b 2053 7061 6365 0d0a 486f 6d65  work Space..Home
-00000060: 2d70 6167 653a 2068 7474 7073 3a2f 2f67  -page: https://g
-00000070: 6974 6875 622e 636f 6d2f 4d6f 7272 6973  ithub.com/Morris
-00000080: 7379 4c61 622f 634e 4d46 2d53 4e53 0d0a  syLab/cNMF-SNS..
-00000090: 4175 7468 6f72 3a20 5465 6420 5665 7268  Author: Ted Verh
-000000a0: 6579 0d0a 4175 7468 6f72 2d65 6d61 696c  ey..Author-email
-000000b0: 3a20 7462 7665 7268 6579 4075 6361 6c67  : tbverhey@ucalg
-000000c0: 6172 792e 6361 0d0a 5072 6f6a 6563 742d  ary.ca..Project-
-000000d0: 5552 4c3a 2042 7567 2054 7261 636b 6572  URL: Bug Tracker
-000000e0: 2c20 6874 7470 733a 2f2f 6769 7468 7562  , https://github
-000000f0: 2e63 6f6d 2f4d 6f72 7269 7373 794c 6162  .com/MorrissyLab
-00000100: 2f63 4e4d 462d 534e 532f 6973 7375 6573  /cNMF-SNS/issues
-00000110: 0d0a 436c 6173 7369 6669 6572 3a20 5072  ..Classifier: Pr
-00000120: 6f67 7261 6d6d 696e 6720 4c61 6e67 7561  ogramming Langua
-00000130: 6765 203a 3a20 5079 7468 6f6e 203a 3a20  ge :: Python :: 
-00000140: 330d 0a43 6c61 7373 6966 6965 723a 204c  3..Classifier: L
-00000150: 6963 656e 7365 203a 3a20 4f53 4920 4170  icense :: OSI Ap
-00000160: 7072 6f76 6564 203a 3a20 4d49 5420 4c69  proved :: MIT Li
-00000170: 6365 6e73 650d 0a43 6c61 7373 6966 6965  cense..Classifie
-00000180: 723a 204f 7065 7261 7469 6e67 2053 7973  r: Operating Sys
-00000190: 7465 6d20 3a3a 204f 5320 496e 6465 7065  tem :: OS Indepe
-000001a0: 6e64 656e 740d 0a52 6571 7569 7265 732d  ndent..Requires-
-000001b0: 5079 7468 6f6e 3a20 3e3d 332e 390d 0a44  Python: >=3.9..D
-000001c0: 6573 6372 6970 7469 6f6e 2d43 6f6e 7465  escription-Conte
-000001d0: 6e74 2d54 7970 653a 2074 6578 742f 6d61  nt-Type: text/ma
-000001e0: 726b 646f 776e 0d0a 4c69 6365 6e73 652d  rkdown..License-
-000001f0: 4669 6c65 3a20 4c49 4345 4e53 450d 0a0d  File: LICENSE...
-00000200: 0a0d 0a21 5b63 4e4d 462d 534e 5320 6c6f  ...![cNMF-SNS lo
-00000210: 676f 5d28 6c6f 676f 2e70 6e67 290d 0a0d  go](logo.png)...
-00000220: 0a2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  .---------------
-00000230: 2d2d 0d0a 0d0a 2320 634e 4d46 2d53 4e53  --....# cNMF-SNS
-00000240: 3a20 706f 7765 7266 756c 2066 6163 746f  : powerful facto
-00000250: 7269 7a61 7469 6f6e 2d62 6173 6564 206d  rization-based m
-00000260: 756c 7469 2d6f 6d69 6373 2069 6e74 6567  ulti-omics integ
-00000270: 7261 7469 6f6e 2074 6f6f 6c6b 6974 0d0a  ration toolkit..
-00000280: 0d0a 215b 7665 7273 696f 6e20 6261 6467  ..![version badg
-00000290: 655d 2868 7474 7073 3a2f 2f69 6d67 2e73  e](https://img.s
-000002a0: 6869 656c 6473 2e69 6f2f 6261 6467 652f  hields.io/badge/
-000002b0: 7665 7273 696f 6e2d 312e 342e 312d 626c  version-1.4.1-bl
-000002c0: 7565 290d 0a5b 215b 5079 5049 204c 6174  ue)..[![PyPI Lat
-000002d0: 6573 7420 5265 6c65 6173 655d 2868 7474  est Release](htt
-000002e0: 7073 3a2f 2f69 6d67 2e73 6869 656c 6473  ps://img.shields
-000002f0: 2e69 6f2f 7079 7069 2f76 2f63 6e6d 6673  .io/pypi/v/cnmfs
-00000300: 6e73 2e73 7667 295d 2868 7474 7073 3a2f  ns.svg)](https:/
-00000310: 2f70 7970 692e 6f72 672f 7072 6f6a 6563  /pypi.org/projec
-00000320: 742f 636e 6d66 736e 732f 290d 0a5b 215b  t/cnmfsns/)..[![
-00000330: 436f 6e64 6120 4c61 7465 7374 2052 656c  Conda Latest Rel
-00000340: 6561 7365 5d28 6874 7470 733a 2f2f 696d  ease](https://im
-00000350: 672e 7368 6965 6c64 732e 696f 2f63 6f6e  g.shields.io/con
-00000360: 6461 2f76 6e2f 636f 6e64 612d 666f 7267  da/vn/conda-forg
-00000370: 652f 636e 6d66 736e 7329 5d28 6874 7470  e/cnmfsns)](http
-00000380: 733a 2f2f 616e 6163 6f6e 6461 2e6f 7267  s://anaconda.org
-00000390: 2f63 6f6e 6461 2d66 6f72 6765 2f63 6e6d  /conda-forge/cnm
-000003a0: 6673 6e73 2f29 0d0a 5b21 5b44 6f63 756d  fsns/)..[![Docum
-000003b0: 656e 7461 7469 6f6e 2073 7461 7475 735d  entation status]
-000003c0: 2868 7474 7073 3a2f 2f72 6561 6474 6865  (https://readthe
-000003d0: 646f 6373 2e6f 7267 2f70 726f 6a65 6374  docs.org/project
-000003e0: 732f 636e 6d66 2d73 6e73 2f62 6164 6765  s/cnmf-sns/badge
-000003f0: 2f3f 7665 7273 696f 6e3d 6c61 7465 7374  /?version=latest
-00000400: 2673 7479 6c65 3d66 6c61 7429 5d28 290d  &style=flat)]().
-00000410: 0a5b 215b 446f 776e 6c6f 6164 735d 2868  .[![Downloads](h
-00000420: 7474 7073 3a2f 2f73 7461 7469 632e 7065  ttps://static.pe
-00000430: 7079 2e74 6563 682f 6261 6467 652f 636e  py.tech/badge/cn
-00000440: 6d66 736e 7329 5d28 6874 7470 733a 2f2f  mfsns)](https://
-00000450: 7065 7079 2e74 6563 682f 7072 6f6a 6563  pepy.tech/projec
-00000460: 742f 636e 6d66 736e 7329 0d0a 5b21 5b4c  t/cnmfsns)..[![L
-00000470: 6963 656e 7365 5d28 6874 7470 733a 2f2f  icense](https://
-00000480: 696d 672e 7368 6965 6c64 732e 696f 2f70  img.shields.io/p
-00000490: 7970 692f 6c2f 636e 6d66 736e 732e 7376  ypi/l/cnmfsns.sv
-000004a0: 6729 5d28 6874 7470 733a 2f2f 6769 7468  g)](https://gith
-000004b0: 7562 2e63 6f6d 2f4d 6f72 7269 7373 794c  ub.com/MorrissyL
-000004c0: 6162 2f63 4e4d 462d 534e 532f 626c 6f62  ab/cNMF-SNS/blob
-000004d0: 2f6d 6169 6e2f 4c49 4345 4e53 4529 0d0a  /main/LICENSE)..
-000004e0: 0d0a 4175 7468 6f72 733a 205b 5465 6420  ..Authors: [Ted 
-000004f0: 5665 7268 6579 5d28 6874 7470 733a 2f2f  Verhey](https://
-00000500: 6769 7468 7562 2e63 6f6d 2f76 6572 6865  github.com/verhe
-00000510: 7974 6229 2c20 5b48 6565 776f 6e20 5365  ytb), [Heewon Se
-00000520: 6f5d 2868 7474 7073 3a2f 2f67 6974 6875  o](https://githu
-00000530: 622e 636f 6d2f 6c6f 6f74 7069 7a29 2c20  b.com/lootpiz), 
-00000540: 5b53 6f72 616e 6120 4d6f 7272 6973 7379  [Sorana Morrissy
-00000550: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-00000560: 2e63 6f6d 2f61 6e63 6173 6f72 616e 6129  .com/ancasorana)
-00000570: 0d0a 0d0a 2a2a 634e 4d46 2d53 4e53 2a2a  ....**cNMF-SNS**
-00000580: 2028 636f 6e73 656e 7375 7320 4e6f 6e2d   (consensus Non-
-00000590: 6e65 6761 7469 7665 204d 6174 7269 7820  negative Matrix 
-000005a0: 4661 6374 6f72 697a 6174 696f 6e20 536f  Factorization So
-000005b0: 6c75 7469 6f6e 204e 6574 776f 726b 2053  lution Network S
-000005c0: 7061 6365 2920 6973 2061 2050 7974 686f  pace) is a Pytho
-000005d0: 6e20 7061 636b 6167 6520 656e 6162 6c69  n package enabli
-000005e0: 6e67 206d 6f73 6169 6320 696e 7465 6772  ng mosaic integr
-000005f0: 6174 696f 6e20 6f66 2062 756c 6b2c 2073  ation of bulk, s
-00000600: 696e 676c 652d 6365 6c6c 2c20 616e 640d  ingle-cell, and.
-00000610: 0a73 7061 7469 616c 2065 7870 7265 7373  .spatial express
-00000620: 696f 6e20 6461 7461 2062 6574 7765 656e  ion data between
-00000630: 2061 6e64 2077 6974 6869 6e20 6461 7461   and within data
-00000640: 7365 7473 2e20 4461 7461 7365 7473 2063  sets. Datasets c
-00000650: 616e 2068 6176 6520 7061 7274 6961 6c6c  an have partiall
-00000660: 7920 6f76 6572 6c61 7070 696e 6720 6665  y overlapping fe
-00000670: 6174 7572 6573 2028 6567 2e20 6765 6e65  atures (eg. gene
-00000680: 7329 2061 7320 7765 6c6c 2061 7320 6e6f  s) as well as no
-00000690: 6e2d 6f76 6572 6c61 7070 696e 6720 6665  n-overlapping fe
-000006a0: 6174 7572 6573 2e20 634e 4d46 2070 726f  atures. cNMF pro
-000006b0: 7669 6465 7320 6120 2a2a 726f 6275 7374  vides a **robust
-000006c0: 2c20 0d0a 756e 7375 7065 7276 6973 6564  , ..unsupervised
-000006d0: 2a2a 2064 6563 6f6e 766f 6c75 7469 6f6e  ** deconvolution
-000006e0: 206f 6620 6561 6368 2064 6174 6173 6574   of each dataset
-000006f0: 2069 6e74 6f20 6765 6e65 2065 7870 7265   into gene expre
-00000700: 7373 696f 6e20 7072 6f67 7261 6d73 2028  ssion programs (
-00000710: 4745 5073 292e 0d0a 2a2a 4e65 7477 6f72  GEPs)...**Networ
-00000720: 6b2d 6261 7365 6420 696e 7465 6772 6174  k-based integrat
-00000730: 696f 6e2a 2a20 6f66 2047 4550 7320 656e  ion** of GEPs en
-00000740: 6162 6c65 7320 666c 6578 6962 6c65 2069  ables flexible i
-00000750: 6e74 6567 7261 7469 6f6e 206f 6620 6d61  ntegration of ma
-00000760: 6e79 2064 6174 6173 6574 730d 0a61 6372  ny datasets..acr
-00000770: 6f73 7320 6173 7361 7973 2028 6567 2e20  oss assays (eg. 
-00000780: 5072 6f74 6569 6e2c 2052 4e41 2d53 6571  Protein, RNA-Seq
-00000790: 2c20 7363 524e 412d 5365 712c 2073 7061  , scRNA-Seq, spa
-000007a0: 7469 616c 2065 7870 7265 7373 696f 6e29  tial expression)
-000007b0: 2061 6e64 2070 6174 6965 6e74 2063 6f68   and patient coh
-000007c0: 6f72 7473 2e0d 0a0d 0a43 6f6d 6d75 6e69  orts.....Communi
-000007d0: 7469 6573 2077 6974 6820 4745 5073 2066  ties with GEPs f
-000007e0: 726f 6d20 6d75 6c74 6970 6c65 2064 6174  rom multiple dat
-000007f0: 6173 6574 7320 6361 6e20 6265 2061 6e6e  asets can be ann
-00000800: 6f74 6174 6564 2077 6974 6820 6461 7461  otated with data
-00000810: 7365 742d 7370 6563 6966 6963 0d0a 616e  set-specific..an
-00000820: 6e6f 7461 7469 6f6e 7320 746f 2066 6163  notations to fac
-00000830: 696c 6974 6174 6520 696e 7465 7270 7265  ilitate interpre
-00000840: 7461 7469 6f6e 2e0d 0a0d 0a23 2320 e29a  tation.....## ..
-00000850: a14d 6169 6e20 4665 6174 7572 6573 0d0a  .Main Features..
-00000860: 0d0a 4865 7265 2061 7265 206a 7573 7420  ..Here are just 
-00000870: 6120 6665 7720 6f66 2074 6865 2074 6869  a few of the thi
-00000880: 6e67 7320 7468 6174 2063 4e4d 462d 534e  ngs that cNMF-SN
-00000890: 5320 646f 6573 2077 656c 6c3a 0d0a 0d0a  S does well:....
-000008a0: 2d20 4964 656e 7469 6669 6573 2069 6e74  - Identifies int
-000008b0: 6572 7072 6574 6162 6c65 2c20 6164 6469  erpretable, addi
-000008c0: 7469 7665 206e 6f6e 2d6e 6567 6174 6976  tive non-negativ
-000008d0: 6520 6765 6e65 2065 7870 7265 7373 696f  e gene expressio
-000008e0: 6e20 7072 6f67 7261 6d73 0d0a 2d20 4d6f  n programs..- Mo
-000008f0: 7361 6963 2069 6e74 6567 7261 7469 6f6e  saic integration
-00000900: 2064 6f65 7320 6e6f 7420 7265 7175 6972   does not requir
-00000910: 6520 7375 6273 6574 7469 6e67 2066 6561  e subsetting fea
-00000920: 7475 7265 732f 6765 6e65 7320 746f 0d0a  tures/genes to..
-00000930: 2020 6120 7368 6172 6564 206f 7220 6f76    a shared or ov
-00000940: 6572 6469 7370 6572 7365 6420 7375 6273  erdispersed subs
-00000950: 6574 0d0a 2d20 4964 6561 6c20 666f 7220  et..- Ideal for 
-00000960: 696e 6372 656d 656e 7461 6c20 696e 7465  incremental inte
-00000970: 6772 6174 696f 6e20 2861 6464 696e 6720  gration (adding 
-00000980: 6461 7461 7365 7473 206f 6e65 2061 7420  datasets one at 
-00000990: 6120 7469 6d65 2920 7369 6e63 650d 0a20  a time) since.. 
-000009a0: 2064 6563 6f6e 766f 6c75 7469 6f6e 2069   deconvolution i
-000009b0: 7320 7065 7266 6f72 6d65 6420 696e 6465  s performed inde
-000009c0: 7065 6e64 656e 746c 7920 6f6e 2065 6163  pendently on eac
-000009d0: 6820 6461 7461 7365 7420 6765 6e65 7261  h dataset genera
-000009e0: 7469 6e67 2069 6e76 6172 6961 6e74 2047  ting invariant G
-000009f0: 4550 730d 0a2d 2049 6e74 6567 7261 7469  EPs..- Integrati
-00000a00: 6f6e 2064 6f65 7320 6e6f 7420 7265 7175  on does not requ
-00000a10: 6972 6520 636f 6d70 6172 6162 6c65 2064  ire comparable d
-00000a20: 6174 6120 7370 6172 7369 7479 206f 7220  ata sparsity or 
-00000a30: 7365 7175 656e 6369 6e67 2064 6570 7468  sequencing depth
-00000a40: 2028 7369 6e67 6c65 2d63 656c 6c2c 206d   (single-cell, m
-00000a50: 6963 726f 6172 7261 792c 2062 756c 6b29  icroarray, bulk)
-00000a60: 0d0a 2d20 5477 6f20 696e 7465 7266 6163  ..- Two interfac
-00000a70: 6573 3a20 636f 6d6d 616e 642d 6c69 6e65  es: command-line
-00000a80: 2069 6e74 6572 6661 6365 2066 6f72 2072   interface for r
-00000a90: 6170 6964 2064 6174 6120 6578 706c 6f72  apid data explor
-00000aa0: 6174 696f 6e20 616e 6420 7079 7468 6f6e  ation and python
-00000ab0: 0d0a 2020 696e 7465 7266 6163 6520 666f  ..  interface fo
-00000ac0: 7220 6578 7465 6e73 6962 696c 6974 7920  r extensibility 
-00000ad0: 616e 6420 666c 6578 6962 696c 6974 790d  and flexibility.
-00000ae0: 0a0d 0a23 2320 f09f 94a7 2049 6e73 7461  ...## .... Insta
-00000af0: 6c6c 0d0a 0d0a 2323 2320 e298 81ef b88f  ll....### ......
-00000b00: 2050 7562 6c69 6320 5265 6c65 6173 650d   Public Release.
-00000b10: 0a0d 0a49 6e73 7461 6c6c 2074 6865 2070  ...Install the p
-00000b20: 6163 6b61 6765 2077 6974 6820 636f 6e64  ackage with cond
-00000b30: 613a 0d0a 6060 6062 6173 680d 0a63 6f6e  a:..```bash..con
-00000b40: 6461 2069 6e73 7461 6c6c 202d 6320 636f  da install -c co
-00000b50: 6e64 612d 666f 7267 6520 636e 6d66 736e  nda-forge cnmfsn
-00000b60: 730d 0a60 6060 0d0a 0d0a 2323 2320 e29c  s..```....### ..
-00000b70: a820 4c61 7465 7374 2076 6572 7369 6f6e  . Latest version
-00000b80: 2066 726f 6d20 4769 7448 7562 0d0a 0d0a   from GitHub....
-00000b90: 4265 666f 7265 2069 6e73 7461 6c6c 696e  Before installin
-00000ba0: 6720 634e 4d46 2d53 4e53 2075 7369 6e67  g cNMF-SNS using
-00000bb0: 2070 6970 2c20 6974 2069 7320 7265 636f   pip, it is reco
-00000bc0: 6d6d 656e 6465 6420 746f 2066 6972 7374  mmended to first
-00000bd0: 2073 6574 2075 7020 6120 7365 7061 7261   set up a separa
-00000be0: 7465 2063 6f6e 6461 2065 6e76 6972 6f6e  te conda environ
-00000bf0: 6d65 6e74 2061 6e64 2068 6176 6520 636f  ment and have co
-00000c00: 6e64 6120 6d61 6e61 6765 2061 7320 6d61  nda manage as ma
-00000c10: 6e79 2064 6570 656e 6465 6e63 6965 7320  ny dependencies 
-00000c20: 6173 2070 6f73 7369 626c 652e 0d0a 0d0a  as possible.....
-00000c30: 6060 6062 6173 680d 0a63 6f6e 6461 2063  ```bash..conda c
-00000c40: 7265 6174 6520 2d2d 6e61 6d65 2063 6e6d  reate --name cnm
-00000c50: 6673 6e73 202d 6320 636f 6e64 612d 666f  fsns -c conda-fo
-00000c60: 7267 6520 7079 7468 6f6e 3d33 2e31 3020  rge python=3.10 
-00000c70: 616e 6e64 6174 6120 7061 6e64 6173 206e  anndata pandas n
-00000c80: 756d 7079 2073 6369 7079 206d 6174 706c  umpy scipy matpl
-00000c90: 6f74 6c69 6220 7570 7365 7470 6c6f 7420  otlib upsetplot 
-00000ca0: 6874 7470 6c69 6232 2074 6f6d 6c69 2074  httplib2 tomli t
-00000cb0: 6f6d 6c69 2d77 2063 6c69 636b 2070 7967  omli-w click pyg
-00000cc0: 7261 7068 7669 7a20 7079 7468 6f6e 2d69  raphviz python-i
-00000cd0: 6772 6170 6820 7365 6d61 6e74 6963 5f76  graph semantic_v
-00000ce0: 6572 7369 6f6e 2070 7979 616d 6c20 7363  ersion pyyaml sc
-00000cf0: 696b 6974 2d6c 6561 726e 2066 6173 7463  ikit-learn fastc
-00000d00: 6c75 7374 6572 2073 6361 6e70 7920 7079  luster scanpy py
-00000d10: 7961 6d6c 0d0a 636f 6e64 6120 6163 7469  yaml..conda acti
-00000d20: 7661 7465 2063 6e6d 6673 6e73 0d0a 7069  vate cnmfsns..pi
-00000d30: 7020 696e 7374 616c 6c20 6769 742b 6874  p install git+ht
-00000d40: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-00000d50: 2f4d 6f72 7269 7373 794c 6162 2f63 4e4d  /MorrissyLab/cNM
-00000d60: 462d 534e 532e 6769 740d 0a60 6060 0d0a  F-SNS.git..```..
-00000d70: 0d0a 2323 20f0 9f93 9620 446f 6375 6d65  ..## .... Docume
-00000d80: 6e74 6174 696f 6e0d 0a0d 0a23 2323 20f0  ntation....### .
-00000d90: 9f93 9320 5079 7468 6f6e 2069 6e74 6572  ... Python inter
-00000da0: 6661 6365 0d0a 0d0a 546f 2067 6574 2073  face....To get s
-00000db0: 7461 7274 6564 2c20 7361 6d70 6c65 2070  tarted, sample p
-00000dc0: 726f 7465 6f6d 6963 7320 6461 7461 7365  roteomics datase
-00000dd0: 7473 2061 6e64 2061 204a 7570 7974 6572  ts and a Jupyter
-00000de0: 206e 6f74 6562 6f6f 6b20 7475 746f 7269   notebook tutori
-00000df0: 616c 2069 7320 6176 6169 6c61 626c 6520  al is available 
-00000e00: 5b68 6572 655d 282f 7475 746f 7269 616c  [here](/tutorial
-00000e10: 2f74 7574 6f72 6961 6c2e 6970 796e 6229  /tutorial.ipynb)
-00000e20: 2e0d 0a0d 0a44 6574 6169 6c65 6420 4150  .....Detailed AP
-00000e30: 4920 7265 6665 7265 6e63 6520 6361 6e20  I reference can 
-00000e40: 6265 2066 6f75 6e64 206f 6e20 5b52 6561  be found on [Rea
-00000e50: 6454 6865 446f 6373 5d28 6874 7470 733a  dTheDocs](https:
-00000e60: 2f2f 636e 6d66 2d73 6e73 2e72 6561 6474  //cnmf-sns.readt
-00000e70: 6865 646f 6373 2e69 6f2f 292e 0d0a 0d0a  hedocs.io/).....
-00000e80: 0d0a 2323 2320 e28c a8ef b88f 2043 6f6d  ..### ...... Com
-00000e90: 6d61 6e64 206c 696e 6520 696e 7465 7266  mand line interf
-00000ea0: 6163 650d 0a0d 0a53 6565 2074 6865 205b  ace....See the [
-00000eb0: 636f 6d6d 616e 6420 6c69 6e65 2069 6e74  command line int
-00000ec0: 6572 6661 6365 2064 6f63 756d 656e 7461  erface documenta
-00000ed0: 7469 6f6e 5d28 2f43 4c49 2e6d 6429 2e0d  tion](/CLI.md)..
-00000ee0: 0a0d 0a23 2320 f09f 92ad 2047 6574 7469  ...## .... Getti
-00000ef0: 6e67 2048 656c 700d 0a0d 0a46 6f72 2065  ng Help....For e
-00000f00: 7272 6f72 7320 6172 6973 696e 6720 6475  rrors arising du
-00000f10: 7269 6e67 2075 7365 206f 6620 634e 4d46  ring use of cNMF
-00000f20: 2d53 4e53 2c20 6372 6561 7465 2061 6e64  -SNS, create and
-00000f30: 2062 726f 7773 6520 6973 7375 6573 2069   browse issues i
-00000f40: 6e20 7468 6520 5b47 6974 4875 6220 2269  n the [GitHub "i
-00000f50: 7373 7565 7322 2074 6162 5d28 6874 7470  ssues" tab](http
-00000f60: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f4d  s://github.com/M
-00000f70: 6f72 7269 7373 794c 6162 2f63 4e4d 462d  orrissyLab/cNMF-
-00000f80: 534e 532f 6973 7375 6573 292e 0d0a       SNS/issues)...
+00000000: 0d0a 215b 634e 4d46 2d53 4e53 206c 6f67  ..![cNMF-SNS log
+00000010: 6f5d 286c 6f67 6f2e 706e 6729 0d0a 0d0a  o](logo.png)....
+00000020: 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d 2d2d  ----------------
+00000030: 2d0d 0a0d 0a23 2063 4e4d 462d 534e 533a  -....# cNMF-SNS:
+00000040: 2070 6f77 6572 6675 6c20 6661 6374 6f72   powerful factor
+00000050: 697a 6174 696f 6e2d 6261 7365 6420 6d75  ization-based mu
+00000060: 6c74 692d 6f6d 6963 7320 696e 7465 6772  lti-omics integr
+00000070: 6174 696f 6e20 746f 6f6c 6b69 740d 0a0d  ation toolkit...
+00000080: 0a21 5b76 6572 7369 6f6e 2062 6164 6765  .![version badge
+00000090: 5d28 6874 7470 733a 2f2f 696d 672e 7368  ](https://img.sh
+000000a0: 6965 6c64 732e 696f 2f62 6164 6765 2f76  ields.io/badge/v
+000000b0: 6572 7369 6f6e 2d31 2e35 2e30 2d62 6c75  ersion-1.5.0-blu
+000000c0: 6529 0d0a 5b21 5b50 7950 4920 4c61 7465  e)..[![PyPI Late
+000000d0: 7374 2052 656c 6561 7365 5d28 6874 7470  st Release](http
+000000e0: 733a 2f2f 696d 672e 7368 6965 6c64 732e  s://img.shields.
+000000f0: 696f 2f70 7970 692f 762f 636e 6d66 736e  io/pypi/v/cnmfsn
+00000100: 732e 7376 6729 5d28 6874 7470 733a 2f2f  s.svg)](https://
+00000110: 7079 7069 2e6f 7267 2f70 726f 6a65 6374  pypi.org/project
+00000120: 2f63 6e6d 6673 6e73 2f29 0d0a 5b21 5b43  /cnmfsns/)..[![C
+00000130: 6f6e 6461 204c 6174 6573 7420 5265 6c65  onda Latest Rele
+00000140: 6173 655d 2868 7474 7073 3a2f 2f69 6d67  ase](https://img
+00000150: 2e73 6869 656c 6473 2e69 6f2f 636f 6e64  .shields.io/cond
+00000160: 612f 766e 2f63 6f6e 6461 2d66 6f72 6765  a/vn/conda-forge
+00000170: 2f63 6e6d 6673 6e73 295d 2868 7474 7073  /cnmfsns)](https
+00000180: 3a2f 2f61 6e61 636f 6e64 612e 6f72 672f  ://anaconda.org/
+00000190: 636f 6e64 612d 666f 7267 652f 636e 6d66  conda-forge/cnmf
+000001a0: 736e 732f 290d 0a5b 215b 446f 6375 6d65  sns/)..[![Docume
+000001b0: 6e74 6174 696f 6e20 7374 6174 7573 5d28  ntation status](
+000001c0: 6874 7470 733a 2f2f 7265 6164 7468 6564  https://readthed
+000001d0: 6f63 732e 6f72 672f 7072 6f6a 6563 7473  ocs.org/projects
+000001e0: 2f63 6e6d 662d 736e 732f 6261 6467 652f  /cnmf-sns/badge/
+000001f0: 3f76 6572 7369 6f6e 3d6c 6174 6573 7426  ?version=latest&
+00000200: 7374 796c 653d 666c 6174 295d 2829 0d0a  style=flat)]()..
+00000210: 5b21 5b44 6f77 6e6c 6f61 6473 5d28 6874  [![Downloads](ht
+00000220: 7470 733a 2f2f 7374 6174 6963 2e70 6570  tps://static.pep
+00000230: 792e 7465 6368 2f62 6164 6765 2f63 6e6d  y.tech/badge/cnm
+00000240: 6673 6e73 295d 2868 7474 7073 3a2f 2f70  fsns)](https://p
+00000250: 6570 792e 7465 6368 2f70 726f 6a65 6374  epy.tech/project
+00000260: 2f63 6e6d 6673 6e73 290d 0a5b 215b 4c69  /cnmfsns)..[![Li
+00000270: 6365 6e73 655d 2868 7474 7073 3a2f 2f69  cense](https://i
+00000280: 6d67 2e73 6869 656c 6473 2e69 6f2f 7079  mg.shields.io/py
+00000290: 7069 2f6c 2f63 6e6d 6673 6e73 2e73 7667  pi/l/cnmfsns.svg
+000002a0: 295d 2868 7474 7073 3a2f 2f67 6974 6875  )](https://githu
+000002b0: 622e 636f 6d2f 4d6f 7272 6973 7379 4c61  b.com/MorrissyLa
+000002c0: 622f 634e 4d46 2d53 4e53 2f62 6c6f 622f  b/cNMF-SNS/blob/
+000002d0: 6d61 696e 2f4c 4943 454e 5345 290d 0a0d  main/LICENSE)...
+000002e0: 0a41 7574 686f 7273 3a20 5b54 6564 2056  .Authors: [Ted V
+000002f0: 6572 6865 795d 2868 7474 7073 3a2f 2f67  erhey](https://g
+00000300: 6974 6875 622e 636f 6d2f 7665 7268 6579  ithub.com/verhey
+00000310: 7462 292c 205b 4865 6577 6f6e 2053 656f  tb), [Heewon Seo
+00000320: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+00000330: 2e63 6f6d 2f6c 6f6f 7470 697a 292c 205b  .com/lootpiz), [
+00000340: 536f 7261 6e61 204d 6f72 7269 7373 795d  Sorana Morrissy]
+00000350: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+00000360: 636f 6d2f 616e 6361 736f 7261 6e61 290d  com/ancasorana).
+00000370: 0a0d 0a2a 2a63 4e4d 462d 534e 532a 2a20  ...**cNMF-SNS** 
+00000380: 2863 6f6e 7365 6e73 7573 204e 6f6e 2d6e  (consensus Non-n
+00000390: 6567 6174 6976 6520 4d61 7472 6978 2046  egative Matrix F
+000003a0: 6163 746f 7269 7a61 7469 6f6e 2053 6f6c  actorization Sol
+000003b0: 7574 696f 6e20 4e65 7477 6f72 6b20 5370  ution Network Sp
+000003c0: 6163 6529 2069 7320 6120 5079 7468 6f6e  ace) is a Python
+000003d0: 2070 6163 6b61 6765 2065 6e61 626c 696e   package enablin
+000003e0: 6720 6d6f 7361 6963 2069 6e74 6567 7261  g mosaic integra
+000003f0: 7469 6f6e 206f 6620 6275 6c6b 2c20 7369  tion of bulk, si
+00000400: 6e67 6c65 2d63 656c 6c2c 2061 6e64 0d0a  ngle-cell, and..
+00000410: 7370 6174 6961 6c20 6578 7072 6573 7369  spatial expressi
+00000420: 6f6e 2064 6174 6120 6265 7477 6565 6e20  on data between 
+00000430: 616e 6420 7769 7468 696e 2064 6174 6173  and within datas
+00000440: 6574 732e 2044 6174 6173 6574 7320 6361  ets. Datasets ca
+00000450: 6e20 6861 7665 2070 6172 7469 616c 6c79  n have partially
+00000460: 206f 7665 726c 6170 7069 6e67 2066 6561   overlapping fea
+00000470: 7475 7265 7320 2865 672e 2067 656e 6573  tures (eg. genes
+00000480: 2920 6173 2077 656c 6c20 6173 206e 6f6e  ) as well as non
+00000490: 2d6f 7665 726c 6170 7069 6e67 2066 6561  -overlapping fea
+000004a0: 7475 7265 732e 2063 4e4d 4620 7072 6f76  tures. cNMF prov
+000004b0: 6964 6573 2061 202a 2a72 6f62 7573 742c  ides a **robust,
+000004c0: 200d 0a75 6e73 7570 6572 7669 7365 642a   ..unsupervised*
+000004d0: 2a20 6465 636f 6e76 6f6c 7574 696f 6e20  * deconvolution 
+000004e0: 6f66 2065 6163 6820 6461 7461 7365 7420  of each dataset 
+000004f0: 696e 746f 2067 656e 6520 6578 7072 6573  into gene expres
+00000500: 7369 6f6e 2070 726f 6772 616d 7320 2847  sion programs (G
+00000510: 4550 7329 2e0d 0a2a 2a4e 6574 776f 726b  EPs)...**Network
+00000520: 2d62 6173 6564 2069 6e74 6567 7261 7469  -based integrati
+00000530: 6f6e 2a2a 206f 6620 4745 5073 2065 6e61  on** of GEPs ena
+00000540: 626c 6573 2066 6c65 7869 626c 6520 696e  bles flexible in
+00000550: 7465 6772 6174 696f 6e20 6f66 206d 616e  tegration of man
+00000560: 7920 6461 7461 7365 7473 0d0a 6163 726f  y datasets..acro
+00000570: 7373 2061 7373 6179 7320 2865 672e 2050  ss assays (eg. P
+00000580: 726f 7465 696e 2c20 524e 412d 5365 712c  rotein, RNA-Seq,
+00000590: 2073 6352 4e41 2d53 6571 2c20 7370 6174   scRNA-Seq, spat
+000005a0: 6961 6c20 6578 7072 6573 7369 6f6e 2920  ial expression) 
+000005b0: 616e 6420 7061 7469 656e 7420 636f 686f  and patient coho
+000005c0: 7274 732e 0d0a 0d0a 436f 6d6d 756e 6974  rts.....Communit
+000005d0: 6965 7320 7769 7468 2047 4550 7320 6672  ies with GEPs fr
+000005e0: 6f6d 206d 756c 7469 706c 6520 6461 7461  om multiple data
+000005f0: 7365 7473 2063 616e 2062 6520 616e 6e6f  sets can be anno
+00000600: 7461 7465 6420 7769 7468 2064 6174 6173  tated with datas
+00000610: 6574 2d73 7065 6369 6669 630d 0a61 6e6e  et-specific..ann
+00000620: 6f74 6174 696f 6e73 2074 6f20 6661 6369  otations to faci
+00000630: 6c69 7461 7465 2069 6e74 6572 7072 6574  litate interpret
+00000640: 6174 696f 6e2e 0d0a 0d0a 2323 20e2 9aa1  ation.....## ...
+00000650: 4d61 696e 2046 6561 7475 7265 730d 0a0d  Main Features...
+00000660: 0a48 6572 6520 6172 6520 6a75 7374 2061  .Here are just a
+00000670: 2066 6577 206f 6620 7468 6520 7468 696e   few of the thin
+00000680: 6773 2074 6861 7420 634e 4d46 2d53 4e53  gs that cNMF-SNS
+00000690: 2064 6f65 7320 7765 6c6c 3a0d 0a0d 0a2d   does well:....-
+000006a0: 2049 6465 6e74 6966 6965 7320 696e 7465   Identifies inte
+000006b0: 7270 7265 7461 626c 652c 2061 6464 6974  rpretable, addit
+000006c0: 6976 6520 6e6f 6e2d 6e65 6761 7469 7665  ive non-negative
+000006d0: 2067 656e 6520 6578 7072 6573 7369 6f6e   gene expression
+000006e0: 2070 726f 6772 616d 730d 0a2d 204d 6f73   programs..- Mos
+000006f0: 6169 6320 696e 7465 6772 6174 696f 6e20  aic integration 
+00000700: 646f 6573 206e 6f74 2072 6571 7569 7265  does not require
+00000710: 2073 7562 7365 7474 696e 6720 6665 6174   subsetting feat
+00000720: 7572 6573 2f67 656e 6573 2074 6f0d 0a20  ures/genes to.. 
+00000730: 2061 2073 6861 7265 6420 6f72 206f 7665   a shared or ove
+00000740: 7264 6973 7065 7273 6564 2073 7562 7365  rdispersed subse
+00000750: 740d 0a2d 2049 6465 616c 2066 6f72 2069  t..- Ideal for i
+00000760: 6e63 7265 6d65 6e74 616c 2069 6e74 6567  ncremental integ
+00000770: 7261 7469 6f6e 2028 6164 6469 6e67 2064  ration (adding d
+00000780: 6174 6173 6574 7320 6f6e 6520 6174 2061  atasets one at a
+00000790: 2074 696d 6529 2073 696e 6365 0d0a 2020   time) since..  
+000007a0: 6465 636f 6e76 6f6c 7574 696f 6e20 6973  deconvolution is
+000007b0: 2070 6572 666f 726d 6564 2069 6e64 6570   performed indep
+000007c0: 656e 6465 6e74 6c79 206f 6e20 6561 6368  endently on each
+000007d0: 2064 6174 6173 6574 2067 656e 6572 6174   dataset generat
+000007e0: 696e 6720 696e 7661 7269 616e 7420 4745  ing invariant GE
+000007f0: 5073 0d0a 2d20 496e 7465 6772 6174 696f  Ps..- Integratio
+00000800: 6e20 646f 6573 206e 6f74 2072 6571 7569  n does not requi
+00000810: 7265 2063 6f6d 7061 7261 626c 6520 6461  re comparable da
+00000820: 7461 2073 7061 7273 6974 7920 6f72 2073  ta sparsity or s
+00000830: 6571 7565 6e63 696e 6720 6465 7074 6820  equencing depth 
+00000840: 2873 696e 676c 652d 6365 6c6c 2c20 6d69  (single-cell, mi
+00000850: 6372 6f61 7272 6179 2c20 6275 6c6b 290d  croarray, bulk).
+00000860: 0a2d 2054 776f 2069 6e74 6572 6661 6365  .- Two interface
+00000870: 733a 2063 6f6d 6d61 6e64 2d6c 696e 6520  s: command-line 
+00000880: 696e 7465 7266 6163 6520 666f 7220 7261  interface for ra
+00000890: 7069 6420 6461 7461 2065 7870 6c6f 7261  pid data explora
+000008a0: 7469 6f6e 2061 6e64 2070 7974 686f 6e0d  tion and python.
+000008b0: 0a20 2069 6e74 6572 6661 6365 2066 6f72  .  interface for
+000008c0: 2065 7874 656e 7369 6269 6c69 7479 2061   extensibility a
+000008d0: 6e64 2066 6c65 7869 6269 6c69 7479 0d0a  nd flexibility..
+000008e0: 0d0a 2323 20f0 9f94 a720 496e 7374 616c  ..## .... Instal
+000008f0: 6c0d 0a0d 0a23 2323 20e2 9881 efb8 8f20  l....### ...... 
+00000900: 5075 626c 6963 2052 656c 6561 7365 0d0a  Public Release..
+00000910: 0d0a 496e 7374 616c 6c20 7468 6520 7061  ..Install the pa
+00000920: 636b 6167 6520 7769 7468 2063 6f6e 6461  ckage with conda
+00000930: 3a0d 0a60 6060 6261 7368 0d0a 636f 6e64  :..```bash..cond
+00000940: 6120 696e 7374 616c 6c20 2d63 2063 6f6e  a install -c con
+00000950: 6461 2d66 6f72 6765 2063 6e6d 6673 6e73  da-forge cnmfsns
+00000960: 0d0a 6060 600d 0a0d 0a23 2323 20e2 9ca8  ..```....### ...
+00000970: 204c 6174 6573 7420 7665 7273 696f 6e20   Latest version 
+00000980: 6672 6f6d 2047 6974 4875 620d 0a0d 0a42  from GitHub....B
+00000990: 6566 6f72 6520 696e 7374 616c 6c69 6e67  efore installing
+000009a0: 2063 4e4d 462d 534e 5320 7573 696e 6720   cNMF-SNS using 
+000009b0: 7069 702c 2069 7420 6973 2072 6563 6f6d  pip, it is recom
+000009c0: 6d65 6e64 6564 2074 6f20 6669 7273 7420  mended to first 
+000009d0: 7365 7420 7570 2061 2073 6570 6172 6174  set up a separat
+000009e0: 6520 636f 6e64 6120 656e 7669 726f 6e6d  e conda environm
+000009f0: 656e 7420 616e 6420 6861 7665 2063 6f6e  ent and have con
+00000a00: 6461 206d 616e 6167 6520 6173 206d 616e  da manage as man
+00000a10: 7920 6465 7065 6e64 656e 6369 6573 2061  y dependencies a
+00000a20: 7320 706f 7373 6962 6c65 2e0d 0a0d 0a60  s possible.....`
+00000a30: 6060 6261 7368 0d0a 636f 6e64 6120 6372  ``bash..conda cr
+00000a40: 6561 7465 202d 2d6e 616d 6520 636e 6d66  eate --name cnmf
+00000a50: 736e 7320 2d63 2063 6f6e 6461 2d66 6f72  sns -c conda-for
+00000a60: 6765 2070 7974 686f 6e3d 332e 3130 2061  ge python=3.10 a
+00000a70: 6e6e 6461 7461 2070 616e 6461 7320 6e75  nndata pandas nu
+00000a80: 6d70 7920 7363 6970 7920 6d61 7470 6c6f  mpy scipy matplo
+00000a90: 746c 6962 2075 7073 6574 706c 6f74 2068  tlib upsetplot h
+00000aa0: 7474 706c 6962 3220 746f 6d6c 6920 746f  ttplib2 tomli to
+00000ab0: 6d6c 692d 7720 636c 6963 6b20 7079 6772  mli-w click pygr
+00000ac0: 6170 6876 697a 2070 7974 686f 6e2d 6967  aphviz python-ig
+00000ad0: 7261 7068 2073 656d 616e 7469 635f 7665  raph semantic_ve
+00000ae0: 7273 696f 6e20 7079 7961 6d6c 2073 6369  rsion pyyaml sci
+00000af0: 6b69 742d 6c65 6172 6e20 6661 7374 636c  kit-learn fastcl
+00000b00: 7573 7465 7220 7363 616e 7079 2070 7979  uster scanpy pyy
+00000b10: 616d 6c0d 0a63 6f6e 6461 2061 6374 6976  aml..conda activ
+00000b20: 6174 6520 636e 6d66 736e 730d 0a70 6970  ate cnmfsns..pip
+00000b30: 2069 6e73 7461 6c6c 2067 6974 2b68 7474   install git+htt
+00000b40: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+00000b50: 4d6f 7272 6973 7379 4c61 622f 634e 4d46  MorrissyLab/cNMF
+00000b60: 2d53 4e53 2e67 6974 0d0a 6060 600d 0a0d  -SNS.git..```...
+00000b70: 0a23 2320 f09f 9396 2044 6f63 756d 656e  .## .... Documen
+00000b80: 7461 7469 6f6e 0d0a 0d0a 2323 2320 f09f  tation....### ..
+00000b90: 9790 2044 6174 6120 7265 7175 6972 656d  .. Data requirem
+00000ba0: 656e 7473 0d0a 0d0a 634e 4d46 2d53 4e53  ents....cNMF-SNS
+00000bb0: 2063 616e 2066 6163 746f 7269 7a65 2061   can factorize a
+00000bc0: 2077 6964 6520 7661 7269 6574 7920 6f66   wide variety of
+00000bd0: 2064 6174 6173 6574 732c 2062 7574 2077   datasets, but w
+00000be0: 696c 6c20 776f 726b 206f 7074 696d 616c  ill work optimal
+00000bf0: 6c79 2069 6e20 7468 6573 6520 636f 6e64  ly in these cond
+00000c00: 6974 696f 6e73 3a0d 0a20 202d 2055 7365  itions:..  - Use
+00000c10: 2075 6e74 7261 6e73 666f 726d 6564 2028   untransformed (
+00000c20: 7261 7729 2064 6174 6120 7768 6572 6520  raw) data where 
+00000c30: 706f 7373 6962 6c65 2e20 466f 7220 2873  possible. For (s
+00000c40: 6329 524e 412d 5365 7120 6461 7461 2c20  c)RNA-Seq data, 
+00000c50: 7573 6520 6665 6174 7572 6520 636f 756e  use feature coun
+00000c60: 7473 2c20 6e6f 7420 5450 4d20 6f72 206f  ts, not TPM or o
+00000c70: 7468 6572 206c 6f67 2d74 7261 6e73 666f  ther log-transfo
+00000c80: 726d 6564 2076 616c 7565 732e 0d0a 0d0a  rmed values.....
+00000c90: 0d0a 2323 2320 f09f 9393 2050 7974 686f  ..### .... Pytho
+00000ca0: 6e20 696e 7465 7266 6163 650d 0a0d 0a54  n interface....T
+00000cb0: 6f20 6765 7420 7374 6172 7465 642c 2073  o get started, s
+00000cc0: 616d 706c 6520 7072 6f74 656f 6d69 6373  ample proteomics
+00000cd0: 2064 6174 6173 6574 7320 616e 6420 6120   datasets and a 
+00000ce0: 4a75 7079 7465 7220 6e6f 7465 626f 6f6b  Jupyter notebook
+00000cf0: 2074 7574 6f72 6961 6c20 6973 2061 7661   tutorial is ava
+00000d00: 696c 6162 6c65 205b 6865 7265 5d28 2f74  ilable [here](/t
+00000d10: 7574 6f72 6961 6c2f 7475 746f 7269 616c  utorial/tutorial
+00000d20: 2e69 7079 6e62 292e 0d0a 0d0a 4465 7461  .ipynb).....Deta
+00000d30: 696c 6564 2041 5049 2072 6566 6572 656e  iled API referen
+00000d40: 6365 2063 616e 2062 6520 666f 756e 6420  ce can be found 
+00000d50: 6f6e 205b 5265 6164 5468 6544 6f63 735d  on [ReadTheDocs]
+00000d60: 2868 7474 7073 3a2f 2f63 6e6d 662d 736e  (https://cnmf-sn
+00000d70: 732e 7265 6164 7468 6564 6f63 732e 696f  s.readthedocs.io
+00000d80: 2f29 2e0d 0a0d 0a0d 0a23 2323 20e2 8ca8  /).......### ...
+00000d90: efb8 8f20 436f 6d6d 616e 6420 6c69 6e65  ... Command line
+00000da0: 2069 6e74 6572 6661 6365 0d0a 0d0a 5365   interface....Se
+00000db0: 6520 7468 6520 5b63 6f6d 6d61 6e64 206c  e the [command l
+00000dc0: 696e 6520 696e 7465 7266 6163 6520 646f  ine interface do
+00000dd0: 6375 6d65 6e74 6174 696f 6e5d 282f 434c  cumentation](/CL
+00000de0: 492e 6d64 292e 0d0a 0d0a 2323 20f0 9f92  I.md).....## ...
+00000df0: ad20 4765 7474 696e 6720 4865 6c70 0d0a  . Getting Help..
+00000e00: 0d0a 466f 7220 6572 726f 7273 2061 7269  ..For errors ari
+00000e10: 7369 6e67 2064 7572 696e 6720 7573 6520  sing during use 
+00000e20: 6f66 2063 4e4d 462d 534e 532c 2063 7265  of cNMF-SNS, cre
+00000e30: 6174 6520 616e 6420 6272 6f77 7365 2069  ate and browse i
+00000e40: 7373 7565 7320 696e 2074 6865 205b 4769  ssues in the [Gi
+00000e50: 7448 7562 2022 6973 7375 6573 2220 7461  tHub "issues" ta
+00000e60: 625d 2868 7474 7073 3a2f 2f67 6974 6875  b](https://githu
+00000e70: 622e 636f 6d2f 4d6f 7272 6973 7379 4c61  b.com/MorrissyLa
+00000e80: 622f 634e 4d46 2d53 4e53 2f69 7373 7565  b/cNMF-SNS/issue
+00000e90: 7329 2e                                  s).
```

### Comparing `cnmfsns-1.4.1/setup.cfg` & `cnmfsns-1.5.0/setup.cfg`

 * *Files 13% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 00000000: 5b6d 6574 6164 6174 615d 0d0a 6e61 6d65  [metadata]..name
 00000010: 203d 2063 6e6d 6673 6e73 0d0a 7665 7273   = cnmfsns..vers
-00000020: 696f 6e20 3d20 312e 342e 310d 0a61 7574  ion = 1.4.1..aut
+00000020: 696f 6e20 3d20 312e 352e 300d 0a61 7574  ion = 1.5.0..aut
 00000030: 686f 7220 3d20 5465 6420 5665 7268 6579  hor = Ted Verhey
 00000040: 0d0a 6175 7468 6f72 5f65 6d61 696c 203d  ..author_email =
 00000050: 2074 6276 6572 6865 7940 7563 616c 6761   tbverhey@ucalga
 00000060: 7279 2e63 610d 0a64 6573 6372 6970 7469  ry.ca..descripti
 00000070: 6f6e 203d 2063 4e4d 4620 536f 6c75 7469  on = cNMF Soluti
 00000080: 6f6e 204e 6574 776f 726b 2053 7061 6365  on Network Space
 00000090: 0d0a 6c6f 6e67 5f64 6573 6372 6970 7469  ..long_descripti
@@ -30,32 +30,33 @@
 000001d0: 7374 656d 203a 3a20 4f53 2049 6e64 6570  stem :: OS Indep
 000001e0: 656e 6465 6e74 0d0a 0d0a 5b6f 7074 696f  endent....[optio
 000001f0: 6e73 5d0d 0a70 6163 6b61 6765 5f64 6972  ns]..package_dir
 00000200: 203d 200d 0a09 3d20 7372 630d 0a70 6163   = ...= src..pac
 00000210: 6b61 6765 7320 3d20 6669 6e64 3a0d 0a69  kages = find:..i
 00000220: 6e73 7461 6c6c 5f72 6571 7569 7265 7320  nstall_requires 
 00000230: 3d20 0d0a 0961 6e6e 6461 7461 203e 3d20  = ...anndata >= 
-00000240: 302e 382e 300d 0a09 7061 6e64 6173 0d0a  0.8.0...pandas..
-00000250: 096e 756d 7079 0d0a 0973 6369 7079 203e  .numpy...scipy >
-00000260: 3d20 312e 392e 300d 0a09 6d61 7470 6c6f  = 1.9.0...matplo
-00000270: 746c 6962 0d0a 0975 7073 6574 706c 6f74  tlib...upsetplot
-00000280: 0d0a 0968 7474 706c 6962 320d 0a09 746f  ...httplib2...to
-00000290: 6d6c 690d 0a09 746f 6d6c 692d 770d 0a09  mli...tomli-w...
-000002a0: 6469 7374 696e 6374 6970 790d 0a09 636c  distinctipy...cl
-000002b0: 6963 6b0d 0a09 7079 6772 6170 6876 697a  ick...pygraphviz
-000002c0: 0d0a 0969 6772 6170 680d 0a09 7365 6d61  ...igraph...sema
-000002d0: 6e74 6963 5f76 6572 7369 6f6e 0d0a 0970  ntic_version...p
-000002e0: 7979 616d 6c0d 0a09 7363 696b 6974 2d6c  yyaml...scikit-l
-000002f0: 6561 726e 0d0a 0966 6173 7463 6c75 7374  earn...fastclust
-00000300: 6572 0d0a 0973 6361 6e70 790d 0a70 7974  er...scanpy..pyt
-00000310: 686f 6e5f 7265 7175 6972 6573 203d 203e  hon_requires = >
-00000320: 3d33 2e39 0d0a 0d0a 5b6f 7074 696f 6e73  =3.9....[options
-00000330: 2e70 6163 6b61 6765 732e 6669 6e64 5d0d  .packages.find].
-00000340: 0a77 6865 7265 203d 2073 7263 0d0a 0d0a  .where = src....
-00000350: 5b6f 7074 696f 6e73 2e65 6e74 7279 5f70  [options.entry_p
-00000360: 6f69 6e74 735d 0d0a 636f 6e73 6f6c 655f  oints]..console_
-00000370: 7363 7269 7074 7320 3d20 0d0a 0963 6e6d  scripts = ...cnm
-00000380: 6673 6e73 203d 2063 6e6d 6673 6e73 2e63  fsns = cnmfsns.c
-00000390: 6c69 3a63 6c69 0d0a 0d0a 5b65 6767 5f69  li:cli....[egg_i
-000003a0: 6e66 6f5d 0d0a 7461 675f 6275 696c 6420  nfo]..tag_build 
-000003b0: 3d20 0d0a 7461 675f 6461 7465 203d 2030  = ..tag_date = 0
-000003c0: 0d0a 0d0a                                ....
+00000240: 302e 382e 300d 0a09 7365 6162 6f72 6e0d  0.8.0...seaborn.
+00000250: 0a09 7374 6174 736d 6f64 656c 730d 0a09  ..statsmodels...
+00000260: 6e65 7477 6f72 6b78 0d0a 0970 616e 6461  networkx...panda
+00000270: 730d 0a09 6e75 6d70 790d 0a09 7363 6970  s...numpy...scip
+00000280: 7920 3e3d 2031 2e39 2e30 0d0a 096d 6174  y >= 1.9.0...mat
+00000290: 706c 6f74 6c69 620d 0a09 7570 7365 7470  plotlib...upsetp
+000002a0: 6c6f 740d 0a09 6874 7470 6c69 6232 0d0a  lot...httplib2..
+000002b0: 0974 6f6d 6c69 0d0a 0974 6f6d 6c69 2d77  .tomli...tomli-w
+000002c0: 0d0a 0964 6973 7469 6e63 7469 7079 0d0a  ...distinctipy..
+000002d0: 0963 6c69 636b 0d0a 0970 7967 7261 7068  .click...pygraph
+000002e0: 7669 7a0d 0a09 6967 7261 7068 0d0a 0970  viz...igraph...p
+000002f0: 7979 616d 6c0d 0a09 7363 696b 6974 2d6c  yyaml...scikit-l
+00000300: 6561 726e 0d0a 0966 6173 7463 6c75 7374  earn...fastclust
+00000310: 6572 0d0a 0973 6361 6e70 790d 0a70 7974  er...scanpy..pyt
+00000320: 686f 6e5f 7265 7175 6972 6573 203d 203e  hon_requires = >
+00000330: 3d33 2e39 0d0a 0d0a 5b6f 7074 696f 6e73  =3.9....[options
+00000340: 2e70 6163 6b61 6765 732e 6669 6e64 5d0d  .packages.find].
+00000350: 0a77 6865 7265 203d 2073 7263 0d0a 0d0a  .where = src....
+00000360: 5b6f 7074 696f 6e73 2e65 6e74 7279 5f70  [options.entry_p
+00000370: 6f69 6e74 735d 0d0a 636f 6e73 6f6c 655f  oints]..console_
+00000380: 7363 7269 7074 7320 3d20 0d0a 0963 6e6d  scripts = ...cnm
+00000390: 6673 6e73 203d 2063 6e6d 6673 6e73 2e63  fsns = cnmfsns.c
+000003a0: 6c69 3a63 6c69 0d0a 0d0a 5b65 6767 5f69  li:cli....[egg_i
+000003b0: 6e66 6f5d 0d0a 7461 675f 6275 696c 6420  nfo]..tag_build 
+000003c0: 3d20 0d0a 7461 675f 6461 7465 203d 2030  = ..tag_date = 0
+000003d0: 0d0a 0d0a                                ....
```

### Comparing `cnmfsns-1.4.1/src/cnmfsns/__init__.py` & `cnmfsns-1.5.0/src/cnmfsns/__init__.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns/cli.py` & `cnmfsns-1.5.0/src/cnmfsns/cli.py`

 * *Files 1% similar despite different names*

```diff
@@ -160,15 +160,15 @@
     
     # output text file
     gene_stats = dataset.adata.var
     os.makedirs(os.path.normpath(os.path.join(output_dir, name, "odgenes")), exist_ok=True)
     gene_stats.to_csv(os.path.join(output_dir, name, "odgenes", "genestats.tsv"), sep="\t")
 
     # create mean vs variance plots
-    fig = plot_feature_dispersion(dataset, show_selected=False)["default"]
+    fig = plot_feature_dispersion(dataset, show_selected=False)
     fig.savefig(os.path.join(output_dir, name, "odgenes.pdf"), facecolor='white')
     fig.savefig(os.path.join(output_dir, name, "odgenes.png"), dpi=400, facecolor='white')
 
 
 @click.command(name="set-parameters")
 @click.option(
     "-n", "--name", type=str, required=True, 
@@ -261,16 +261,16 @@
         else:
             method_str = cli_method_str
             threshold = float(odg_param)
         method = {method_str: threshold}
         dataset.select_overdispersed_genes(overdispersion_metric=metric_str, min_mean=min_mean,
                                            **method)
 
-    # create mean vs variance plots, updated with selected genes
-    fig = plot_feature_dispersion(dataset, show_selected=True)["default"]
+    # create mean vs variance plot, updated with selected genes
+    fig = plot_feature_dispersion(dataset, show_selected=True)
     fig.savefig(os.path.join(output_dir, name, "odgenes.pdf"), facecolor='white')
     fig.savefig(os.path.join(output_dir, name, "odgenes.png"), dpi=400, facecolor='white')
 
     # output table with gene overdispersion measures
     dataset.adata.var.to_csv(os.path.join(output_dir, name, "odgenes", "genestats.tsv"), sep="\t")
     
     # process k-value selection inputs
@@ -619,17 +619,19 @@
     
     fig = colors.plot_dataset_colors_legend()
     fig.savefig(os.path.join(sns_output_dir, "dataset_colors_legend.pdf"))
     plt.close(fig)
     
     snsmap.write_communities_toml( os.path.join(sns_output_dir, "communities.toml"))
     pd.DataFrame.from_dict(data=snsmap.gep_communities, orient='index').to_csv(os.path.join(sns_output_dir, 'gep_communities.txt'), sep="\t", header=False)
-        
-    representative_geps = snsmap.get_representative_geps()
-    representative_geps.to_csv(os.path.join(sns_output_dir, "representative_geps.txt"), sep="\t") # outputs the GEPs to text file
+    
+    central_gep_ids = snsmap.get_central_geps()
+    central_geps = snsmap.integration.get_geps()[central_gep_ids.index]
+    central_geps.columns = pd.MultiIndex.from_tuples([[community] + list(gep_id) for community, gep_id in zip(central_gep_ids, central_gep_ids.index)], names=["Community", "dataset", "k", "GEP"])
+    central_geps.to_csv(os.path.join(sns_output_dir, "central_geps.txt"), sep="\t") # outputs the GEPs to text file
 
     logging.info("Creating SNS plots...")
 
     # plot membership of datasets and ranks for each community
     fig = plot_community_by_dataset_rank(snsmap, colors)
     fig.savefig(os.path.join(sns_output_dir, "communities_by_dataset_rank.pdf"))
     fig.savefig(os.path.join(sns_output_dir, "communities_by_dataset_rank.png"), dpi=600)
```

### Comparing `cnmfsns-1.4.1/src/cnmfsns/cnmf.py` & `cnmfsns-1.5.0/src/cnmfsns/cnmf.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns/colors.py` & `cnmfsns-1.5.0/src/cnmfsns/colors.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns/config.py` & `cnmfsns-1.5.0/src/cnmfsns/config.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns/dataset.py` & `cnmfsns-1.5.0/src/cnmfsns/dataset.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns/integration.py` & `cnmfsns-1.5.0/src/cnmfsns/integration.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns/plots.py` & `cnmfsns-1.5.0/src/cnmfsns/plots.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns/sns.py` & `cnmfsns-1.5.0/src/cnmfsns/sns.py`

 * *Files 4% similar despite different names*

```diff
@@ -1,14 +1,14 @@
 
 from .integration import Integration
 from .utils import node_to_gep
 
 
 from collections.abc import Collection, Iterable
-from typing import Union, Optional, Dict, List
+from typing import Union, Optional, Dict, List, Literal
 import logging
 
 import numpy as np
 import pandas as pd
 import networkx as nx
 import matplotlib as mpl
 import igraph
@@ -78,22 +78,27 @@
         """
         nodes = []
         for node in self.gep_graph.nodes:
             dataset_name, k_str, gep_str = node.split("|")
             nodes.append((dataset_name, int(k_str), int(gep_str)))
         return nodes
     
-    def get_community_usage(self, subset_datasets: Optional[Union[str, Iterable[str]]] = None, normalize=True):
+    def get_community_usage(self,
+                            subset_datasets: Optional[Union[str, Iterable[str]]] = None,
+                            normalize: bool = True,
+                            discretize: bool = False):
         """
         Get median usage of each community of GEPs for each samples. 
 
         :param subset_datasets: dataset name or iterable of dataset names to subset the results, defaults to None
         :type subset_datasets: str or Iterable[str], optional
-        :param normalize: Normalize the GEP usage matrix such that for each value of k, usage of all GEPs sums to 1. Defaults to False
+        :param normalize: Normalize the community usage matrix such that for each value of k, usage of all communities of GEPs sums to 1. Defaults to False
         :type normalize: bool, optional
+        :param discretize: Discretizes the community usage matrix such that for each value of k, each sample has usage of only 1 community (the one with the maximum usage). Defaults to False
+        :type discretize: bool, optional
         :return: observations  communities matrix
         :rtype: pd.DataFrame
         """
         usage = self.integration.get_usages(normalize=normalize)
         ic_usage = []
         
         if subset_datasets is None:
@@ -114,14 +119,16 @@
                 data.append(gep_comm.median(axis=1).rename((community, dataset_name)))
             data = pd.concat(data, axis=1).droplevel(axis=1, level=1).dropna(how="all")
             data.columns.rename("Community", inplace=True)
             ic_usage.append(data.sort_index(axis=0))
         ic_usage = pd.concat(ic_usage)
         if normalize:
             ic_usage = ic_usage.div(ic_usage.sum(axis=1), axis=0)
+        if discretize:
+            ic_usage = ic_usage.eq(ic_usage.max(axis=1), axis=0).astype(int)
         return ic_usage
     
     def get_sample_entropy(self, subset_datasets: Optional[Union[str, Iterable]] = None):
         """Get shannon diversity of Community Usage for each sample.
 
         :param subset_datasets: dataset name or iterable of dataset names to subset the results, defaults to None
         :type subset_datasets: str or Iterable[str], optional
@@ -551,71 +558,179 @@
         edge_list = pd.DataFrame(edge_list, columns = ("comm1", "comm2", "n_edges", "weight"))
         self.comm_graph = nx.from_pandas_edgelist(df = pd.DataFrame(edge_list, columns = ("comm1", "comm2", "n_edges", "weight")),
                                                   source = "comm1",
                                                   target = "comm2",
                                                   edge_attr=["n_edges", "weight"])
         self.comm_graph.add_nodes_from(self.communities.keys())
         
-    def get_representative_gep_table(self,
-                                method: str = "min_k",
-                                min_k: int = 2
-                                ) -> pd.DataFrame:
-        """Return a dataframe with GEPs that represent each community. Representative GEPs are the lowest rank GEPs in each community.
-
-        :param method: Method for identifying 'representative' GEPs, defaults to "min_k"
-        :type method: str, optional
-        :param min_k: Minimum k-value for representative GEPs. Can be set higher than 2 to prevent low-resolution results. Defaults to 2.
-        :type min_k: int, optional
-        :return: Representative GEP table
-        :rtype: pd.DataFrame
+    def get_central_geps(self,
+                         correlation_axis: Literal["geps", "usage"] = "geps"
+                         ):
+        """Select GEPs based on correlation with the median of all GEPs in each community
+        
+        :param correlation_axis: axis on which to compute correlations between GEPs whether correlating the GEPs or the usages, defaults to "geps"
+        :type correlation_axis: int or dict
+        :return: Communities, indexed by the most central GEP for each dataset
+        :rtype: pd.Series
         """
-        if method == "min_k":
-            logging.info(f"Selecting representative ranks for each community with min_k = {min_k}")
-            # get minimum k GEPs for each community/dataset combination.
-            table = []
+        selected_geps = {}
+
+        if correlation_axis == "geps":
+            cg = self.consensus()
+            geps_df = self.integration.get_geps()
             for community, nodes in self.communities.items():
-                nodes = pd.DataFrame([n.split("|") for n in nodes], columns=["dataset", "k", "GEP"])
-                nodes = nodes[nodes["k"].astype(int) >= min_k]
                 for dataset_name in self.integration.datasets:
-                    dataset_nodes = nodes[nodes["dataset"] == dataset_name].copy()
-                    if dataset_nodes.shape[0]:
-                        block = dataset_nodes[dataset_nodes["k"].astype(int) == dataset_nodes["k"].astype(int).min()].copy()
-                        block["Community"] = community
-                        table.append(block)
-            table = pd.concat(table).set_index("Community")
-        else:
-            raise ValueError
-        
-        return table
-        
+                    geps = [node_to_gep(node) for node in nodes]
+                    geps = [gep for gep in geps if gep[0] == dataset_name]
+                    if geps:
+                        top_gep = geps_df[geps].corrwith(cg[(community, dataset_name)]).idxmax()
+                        selected_geps[top_gep] = community
+            
 
-    def get_representative_geps(self,
-                                method: str = "min_k",
-                                min_k: int = 2
-                                ) -> pd.DataFrame:
-        """Return a dataframe with GEPs that represent each community. Representative GEPs are the lowest rank GEPs in each community.
+        elif correlation_axis == "usage":
+            cu = self.get_community_usage()
+            usages = self.integration.get_usages(normalize=True)
+            for community, nodes in self.communities.items():
+                for dataset_name in self.integration.datasets:
+                    geps = [node_to_gep(node) for node in nodes]
+                    geps = [gep for gep in geps if gep[0] == dataset_name]
+                    if geps:
+                        top_gep = usages.loc[dataset_name, geps].corrwith(cu.loc[dataset_name, community]).idxmax()
+                        selected_geps[top_gep] = community
+
+        selected_geps = pd.Series(selected_geps, name="Community")
+        selected_geps.index.rename(("dataset", "k", "GEP"), inplace=True)
+        selected_geps = selected_geps.sort_index().sort_values(key=lambda x: x.map(self.ordered_community_names.index))
 
-        :param method: Method for identifying 'representative' GEPs, defaults to "min_k"
-        :type method: str, optional
-        :param min_k: Minimum k-value for representative GEPs. Can be set higher than 2 to prevent low-resolution results. Defaults to 2.
-        :type min_k: int, optional
-        :return: features  GEP table subset for 'representative' GEPs
-        :rtype: pd.DataFrame
+        return selected_geps
+
+    def get_selected_rank_geps(self,
+                            k: Union[int, Dict[str, int]]):
+        """Select GEPs based on rank. k may be either a single value for all datasets
+        as an integer, or as a dict with separate values for each dataset.
+
+        :param k: a rank or dict with dataset keys and rank values, defaults to None
+        :type k: int or dict
+        :return: Communities, indexed by the lowest rank GEP(s) for each dataset
+        :rtype: pd.Series
         """
-        geps = self.integration.get_geps()
-        
-        table = self.get_representative_gep_table(method = method, min_k = min_k)
+        geps = pd.Series(self.gep_communities, name="Community")
+        geps.index = pd.MultiIndex.from_tuples((node_to_gep(node) for node in geps.index), names=["dataset", "k", "GEP"])
+        geps = geps.sort_index()
+        if isinstance(k, int):
+            selected_geps = geps.xs(k, level=1, drop_level=False)
+        elif isinstance(k, dict):
+            selected_geps = []
+            for dataset_name, ds_k in k.items():
+                selected_geps.append(geps.xs(dataset_name, drop_level=False).xs(ds_k, level=1, drop_level=False))
+            selected_geps = pd.concat(selected_geps)
+        selected_geps = selected_geps.sort_index().sort_values(key=lambda x: x.map(self.ordered_community_names.index))
+        return selected_geps
+
+
+    def get_lowest_rank_geps(self,
+                                min_k: Optional[Union[int, Dict[str, int]]] = None,
+                            ) -> pd.Series:
+        """Identify the GEPS that are the lowest rank for each dataset. A minimum rank to be considered may be supplied,
+        either for all datasets as an integer, or as a dict with separate thresholds for each dataset.
+
+        :param min_k: a minimum rank to consider for the minimal-rank GEPs, defaults to None
+        :type min_k: int or dict, optional
+        :return: Communities, indexed by the lowest rank GEP(s) for each dataset
+        :rtype: pd.Series
+        """
+        geps = pd.Series(self.gep_communities, name="Community")
+        geps.index = pd.MultiIndex.from_tuples((node_to_gep(node) for node in geps.index), names=["dataset", "k", "GEP"])
+        geps = geps.sort_index()
+        if isinstance(min_k, int):
+            geps = geps[geps.index.get_level_values(1) >= min_k]
+        elif isinstance(min_k, dict):
+            
+            keep = []
+            for gep in geps.index:
+                if not gep[0] in min_k:
+                    keep.append(True)
+                elif gep[1] >= min_k[gep[0]]:
+                    keep.append(True)
+                else:
+                    keep.append(False)
+            geps = geps[keep]
+
         selected_geps = []
-        for community, gep in table.iterrows():
-            gep = geps[gep['dataset'], int(gep['k']), int(gep['GEP'])]
-            gep = gep.rename((community,) + gep.name)
-            selected_geps.append(gep)
-        selected_geps = pd.concat(selected_geps, axis=1)
-        selected_geps.columns.rename(("community", "dataset", "k", "GEP"), inplace=True)
+        for community, cgeps in geps.groupby(geps):
+            for dataset_name, cdgeps in cgeps.groupby(axis=0, level=0):
+                min_rank = cdgeps.index.get_level_values(1).min()
+                min_rank_geps = cdgeps[cdgeps.index.get_level_values(1) == min_rank]
+                selected_geps.append(min_rank_geps)
+        selected_geps = pd.concat(selected_geps)
+        selected_geps = selected_geps.sort_index().sort_values(key=lambda x: x.map(self.ordered_community_names.index))
         return selected_geps
+
+    def count_intracommunity_edges(self):
+        """Counts edges within each community that are within and between datasets.
+
+        :return: Table with number of edges for each dataset and dataset pair
+        :rtype: pd.DataFrame
+        """
+        edgedata = {}
+        for cname, cmembers in self.communities.items():
+            community_graph = self.gep_graph.subgraph(cmembers)
+            stats = {}
+            for edge in community_graph.edges:
+                datasets = tuple(sorted((edge[0].split("|")[0], edge[1].split("|")[0])))
+                if datasets not in stats:
+                    stats[datasets] = 0
+                else:
+                    stats[datasets] += 1
+            edgedata[cname] = stats
+        edgedata = pd.DataFrame(edgedata).T.fillna(0)
+        edgedata.columns = ["-".join(sorted(items)) for items in edgedata.columns]
+        return edgedata
+
+    def most_correlated_edge_between_datasets(self,
+                                              ds1: str,
+                                              ds2: str,
+                                              ds1_rank: Optional[int] = None,
+                                              ds2_rank: Optional[int] = None):
+
+        """Identifies the most correlated edge between two datasets within each community.
+        Optionally, you can fix the rank of 1 or both datasets to restrict the search space.
+
+        :param ds1: name of dataset 1
+        :type ds1: str
+        :param ds2: name of dataset 2
+        :type ds2: str
+        :param ds1_rank: rank of dataset 1
+        :type ds1_rank: int, optional
+        :param ds2_rank: rank of dataset 2
+        :type ds2_rank: int, optional
+        :return: High-correlation edges, one per community
+        :rtype: pd.DataFrame
+        """
+        selected_edges = []
+        for cname, cmembers in self.communities.items():
+            # get correlation values
+            community_idx = pd.MultiIndex.from_tuples([node_to_gep(node) for node in cmembers])
+            ds1_idx = community_idx[community_idx.get_level_values(0) == ds1]
+            if ds1_rank is not None:
+                ds1_idx = ds1_idx[ds1_idx.get_level_values(1) == ds1_rank]
+            ds2_idx = community_idx[community_idx.get_level_values(0) == ds2]
+            if ds2_rank is not None:
+                ds2_idx = ds2_idx[ds2_idx.get_level_values(1) == ds2_rank]
+            community_corr = self.integration.corr_matrix.loc[ds1_idx, ds2_idx]
+            maxcorr = community_corr[community_corr == community_corr.max().max()]
+            maxcorr = maxcorr.dropna(how="all", axis=0).dropna(how="all", axis=1)
+            maxcorr = maxcorr.melt(ignore_index=False).dropna()
+            maxcorr["Community"] = cname 
+            maxcorr = maxcorr.reset_index().set_index("Community").drop(columns="value")
+            maxcorr.columns = pd.MultiIndex.from_product([(ds1, ds2), ("dataset", "k", "GEP")])
+            selected_edges.append(maxcorr)
+        selected_edges = pd.concat(selected_edges)
+        return selected_edges
+
     
     def consensus(self,
                   method: str = "median",
                   min_k: int = 2
                   ) -> pd.DataFrame:
         """Generate a 'consensus' GEP for each community and dataset by taking the median of all constituent GEPs, separately for each dataset.
```

### Comparing `cnmfsns-1.4.1/src/cnmfsns/utils.py` & `cnmfsns-1.5.0/src/cnmfsns/utils.py`

 * *Files identical despite different names*

### Comparing `cnmfsns-1.4.1/src/cnmfsns.egg-info/PKG-INFO` & `cnmfsns-1.5.0/PKG-INFO`

 * *Files 10% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: cnmfsns
-Version: 1.4.1
+Version: 1.5.0
 Summary: cNMF Solution Network Space
 Home-page: https://github.com/MorrissyLab/cNMF-SNS
 Author: Ted Verhey
 Author-email: tbverhey@ucalgary.ca
 Project-URL: Bug Tracker, https://github.com/MorrissyLab/cNMF-SNS/issues
 Classifier: Programming Language :: Python :: 3
 Classifier: License :: OSI Approved :: MIT License
@@ -16,15 +16,15 @@
 
 ![cNMF-SNS logo](logo.png)
 
 -----------------
 
 # cNMF-SNS: powerful factorization-based multi-omics integration toolkit
 
-![version badge](https://img.shields.io/badge/version-1.4.1-blue)
+![version badge](https://img.shields.io/badge/version-1.5.0-blue)
 [![PyPI Latest Release](https://img.shields.io/pypi/v/cnmfsns.svg)](https://pypi.org/project/cnmfsns/)
 [![Conda Latest Release](https://img.shields.io/conda/vn/conda-forge/cnmfsns)](https://anaconda.org/conda-forge/cnmfsns/)
 [![Documentation status](https://readthedocs.org/projects/cnmf-sns/badge/?version=latest&style=flat)]()
 [![Downloads](https://static.pepy.tech/badge/cnmfsns)](https://pepy.tech/project/cnmfsns)
 [![License](https://img.shields.io/pypi/l/cnmfsns.svg)](https://github.com/MorrissyLab/cNMF-SNS/blob/main/LICENSE)
 
 Authors: [Ted Verhey](https://github.com/verheytb), [Heewon Seo](https://github.com/lootpiz), [Sorana Morrissy](https://github.com/ancasorana)
@@ -68,14 +68,20 @@
 conda create --name cnmfsns -c conda-forge python=3.10 anndata pandas numpy scipy matplotlib upsetplot httplib2 tomli tomli-w click pygraphviz python-igraph semantic_version pyyaml scikit-learn fastcluster scanpy pyyaml
 conda activate cnmfsns
 pip install git+https://github.com/MorrissyLab/cNMF-SNS.git
 ```
 
 ##  Documentation
 
+###  Data requirements
+
+cNMF-SNS can factorize a wide variety of datasets, but will work optimally in these conditions:
+  - Use untransformed (raw) data where possible. For (sc)RNA-Seq data, use feature counts, not TPM or other log-transformed values.
+
+
 ###  Python interface
 
 To get started, sample proteomics datasets and a Jupyter notebook tutorial is available [here](/tutorial/tutorial.ipynb).
 
 Detailed API reference can be found on [ReadTheDocs](https://cnmf-sns.readthedocs.io/).
```

